satpy/readers/nwcsaf_nc.py

```diff
@@ -268,30 +268,33 @@ def get_area_def(self, dsid):
 
         nlines, ncols = self.nc[dsid['name']].shape
 
-        area = self._ensure_area_def_is_in_meters(
-            get_area_def('some_area_name',
-                         "On-the-fly area",
-                         'geosmsg',
-                         proj_str,
-                         ncols,
-                         nlines,
-                         area_extent))
+        crs, area_extent = self._ensure_crs_extents_in_meters(proj_str, area_extent)
+        area = AreaDefinition('some_area_name',
+                              "On-the-fly area",
+                              'geosmsg',
+                              crs,
+                              ncols,
+                              nlines,
+                              area_extent)
 
         return area
 
     @staticmethod
-    def _ensure_area_def_is_in_meters(area_definition):
+    def _ensure_crs_extents_in_meters(proj_str, area_extent):
```

Comment:
I would rather that `_get_projection` would return a CRS and an area extent that get then passed to this function, and let this function do the conversion to proj_dict if necessary.

Reply:
I think I see where you're going with this, but I disagree.

> let this function do the conversion to proj_dict if necessary

I think it is always necessary... :thinking: unless I check the CRS object for the units and then only convert to the proj dict then...

Darn it. I'll try it.

I was going to say: Since the CRS would always have to be converted to a dict then there isn't much benefit to CRS -> dict -> CRS. However, the new implementation of `proj4_str_to_dict` in pyresample actually does this `CRS -> dict` conversion anyway so :shrug:

Reply:
Done.